runtime/src/chpl-file-utils.c

```diff
@@ -77,6 +77,29 @@ qioerr chpl_fs_copy_metadata(const char* source, const char* dest) {
   return err;
 }
 
+char* chpl_fs_curdir(void) {
```

Comment:
A return type of "c_string" has special meaning -- that the underlying string representation is "not owned" by the Chapel compiler.  That means it is permissible in this context to simply return a statically-allocated string.
That being the case, the body of this routine can be collapsed to:
{ return "."; }

The Chapel string implementation deals with whether the low-level string is "owned" or not.  One uses a return type of "c_string_copy" to indicate that Chapel is expected to manage the returned string.

Whether the runtime returns a c_string or a c_string_copy is immaterial -- as long as the right type is chosen in the Chapel extern declaration.  Using c_string for the implementation here is just a recommendation -- because it will use less memory and run faster.